tasks/main.yml

```diff
@@ -1,25 +1,72 @@
 ---
-# tasks file for roles/nginx-ssl-selfsigned
+# tasks file for roles/ssl-certificate
 
-- name: system packages | install openssl
+- name: ssl certificate | install openssl
   become: yes
   yum:
     name: openssl
     state: present
 
-- name: nginx | create certificates directory
+- name: ssl certificate | create certificates directory
   become: yes
   file:
     path: "{{ item | dirname }}"
     state: directory
   with_items:
-    - "{{ nginx_ssl_certificate }}"
-    - "{{ nginx_ssl_certificate_key }}"
+    - "{{ ssl_certificate_path }}"
+    - "{{ ssl_certificate_key_path }}"
 
+# Create from content of variable e.g. from vault
+
+- name: ssl certificate | write SSL certificate
+  become: yes
+  copy:
+    content: "{{ ssl_certificate_content }}"
+    dest: "{{ ssl_certificate_path }}"
+    mode: 0444
+  when: not ssl_certificate_selfsigned_create
+
+- name: ssl certificate | write SSL certificate key
+  become: yes
+  copy:
+    content: "{{ ssl_certificate_key_content }}"
+    dest: "{{ ssl_certificate_key_path }}"
+    mode: 0400
+  no_log: True
+  when: not ssl_certificate_selfsigned_create
+
+# Self-signed
 # http://serialized.net/2013/04/simply-generating-self-signed-ssl-certs-with-ansible/
-- name: nginx | generate self-signed SSL certificate
+
+- name: ssl certificate | generate self-signed SSL certificate
   become: yes
   command:
-    openssl req -new -nodes -x509 -subj "{{ nginx_ssl_certificate_subject }}" -days {{ nginx_ssl_certificate_days }} -keyout {{ nginx_ssl_certificate_key }} -out {{ nginx_ssl_certificate }} -extensions v3_ca
+    openssl req -new -nodes -x509 -subj "{{ ssl_certificate_selfsigned_subject }}" -days {{ ssl_certificate_selfsigned_days }} -keyout {{ ssl_certificate_key_path }} -out {{ ssl_certificate_path }} -extensions v3_ca
   args:
-    creates: "{{ nginx_ssl_certificate }}"
+    # Don't overwrite existing certificate
```

Comment:
:+1: for protecting from overwriting by default typically is a prod server has been configured with SSL certs and a subsequent run is invoked without the private SSL variables.

If I understand correctly the *only* way to overwrite would be to pass an SSL certificate via the `content` variable?

Reply:
Yes